test/upgrade-matrix/nodes.py

```diff
@@ -0,0 +1,86 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
```

Comment:
I'm wondering why this is a new upgrade-matrix test instead of using the already-existing platform checks and extending them. Was it easier this way or less risk of changing anything in the existing tests?

Reply:
Yes, I wanted this as a separate CI job and everything for the following reason:
- it uses randomization, so still needs to prove itself as a stable Nightly job
- the entire concept of going so far back in time to start the upgrade sequence is unproven and I expect more back-and-forth around what upgrades are realistic and permissible.
While this plays out I wanted to keep the existing jobs as they are. Going forward, I will be presenting a formal proposal as to what constitutes a valid upgrade sequence. Once I have this approved by the powers that be, I will rearrange the entire set of upgrade test jobs (that have been added so far in an ad-hoc fashion) to match the spec.